src/sisl/io/sile.py

```diff
@@ -497,6 +500,33 @@ class BaseSile:
     def __init__(self, *args, **kwargs):
         """Just to pass away the args and kwargs"""
 
+    def _sanitize_filename(self, filename) -> Union[Path, ZipPath]:
+        """Sanitize the filename to be a ``Path`` or ``ZipPath`` object.
+
+        If the filename is a ``zipfile.Path``, a ``ZipPath`` or a path with
+        a .zip file in the middle of it, it will be converted to a ``ZipPath``.
+
+        Otherwise, it will be converted to a ``Path`` object.
+
+        Parameters
+        ----------
+        filename :
+            The filename to be sanitized.
+        """
+        if isinstance(filename, zipfile.Path):
+            filename = ZipPath.from_zipfile_path(filename)
+        elif not isinstance(filename, ZipPath):
+            filename = Path(filename)
+
+            # Try to convert to a ZipPath, which will only succeed if there
+            # is a .zip file in the middle of the path
+            try:
+                filename = ZipPath.from_path(filename, self._mode)
+            except FileNotFoundError:
+                pass
+
+        return filename
```

Comment:
it should convert to `Path` here, to follow the signature.

Reply:
It is already converted to `Path` at the beggining of the `elif` and everything should go through that `elif` except the case when filename is a `ZipPath`

Reply:
true, missed that!